web_external/js/app.js

```diff
@@ -1,5 +1,59 @@
 isic.App = girder.App.extend({
 
+    initialize: function (settings) {
+        // Call the parent "initialize", but don't let that call "start()",
+        // since we want to pass custom parameters
+        girder.App.prototype.initialize.call(this, {
+            start: false
+        });
+
+        // Start the app (and fetch the user)
+        this.start({
+            // Don't render automatically, to ensure currentUser is updated
+            // before the first rendering
+            render: false,
+            // Don't start the history, to delay starting navigation until
+            // rendering has occurred
+            history: false
+        }).done(_.bind(function () {
+            // Then update currentUser if necessary
+            this._updateCurrentUser();
+            // Do the things that would normally be done within "start()"
+            this.render();
+            Backbone.history.start({
+                pushState: false
+            });
+        }, this));
+    },
+
+    bindGirderEvents: function () {
+        girder.App.prototype.bindGirderEvents.call(this);
+
+        // It would be better to listen for "g:login.success", as it's triggered
+        // before "g:login", but "g:login.success" isn't triggered after a new
+        // user registers (which is probably a bug)
+
+        // "currentUser" must be updated before all rendering, so unbind the
+        // default binding of "g:login" to "this.login", and ensure that
+        // updating "currentUser" is the event bound to "g:login"; however,
+        // it's an undocumented property of Backbone that events get called in
+        // the order they were bound, so this isn't a great solution
+        girder.events.off('g:login', this.login);
+        girder.events.on('g:login', this._updateCurrentUser(), this);
```

Comment:
This event isn't attached correctly--it's calling `updateCurrentUser()`. Because of this I see console errors if I log out and log in.